src/coreclr/interpreter/compiler.cpp

```diff
@@ -735,7 +735,22 @@ uint32_t InterpCompiler::ConvertOffset(int32_t offset)
     return offset * sizeof(int32_t) + sizeof(void*);
 }
 
-int32_t* InterpCompiler::EmitCodeIns(int32_t *ip, InterpInst *ins, TArray<Reloc*, MemPoolAllocator> *relocs)
+int32_t InterpCompiler::GetFuncletAdjustedVarOffset(int varIndex, bool forFunclet)
+{
+#ifndef FEATURE_REUSE_INTERPRETER_STACK_FOR_NORMAL_FUNCLETS
+    if (forFunclet && m_pVars[varIndex].global)
+    {
+        // In funclets, global vars are accessed relative to the main method frame pointer
+        return -(m_pVars[varIndex].offset + FUNCLET_STACK_ADJUSTMENT_OFFSET);
```

Comment:
This is not an absolute requirement, but if var offsets are no longer actual offsets in certain configurations, I'd like to see `var.offset` become something other than an int, so the type system stops us from using it incorrectly - I want any misuse of `.offset` to fail to compile instead of silently doing the wrong thing.